### PR TITLE
start 1.2.0-develop

### PR DESCRIPTION
**Pull request by kilograham**

### CODE REVIEW

File: pico_sdk_version.cmake [hunk 1]
```diff
@@ -3,10 +3,18 @@
 set(PICO_SDK_VERSION_MAJOR 1)
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
-set(PICO_SDK_VERSION_MINOR 1)
+set(PICO_SDK_VERSION_MINOR 2)
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 set(PICO_SDK_VERSION_REVISION 0)
+# PICO_BUILD_DEFINE: PICO_SDK_VERSION_PRE_RELEASE_ID, optional SDK pre-release version identifier, type=string, pico_base
+# PICO_CONFIG: PICO_SDK_VERSION_PRE_RELEASE_ID, SDK version revision, type=int, pico_base
```
Comment (lurch) [inline pico_sdk_version.cmake:11]: `type=string`? :wink:

Comment (kilograham) [reply pico_sdk_version.cmake]: fixed

File: pico_sdk_version.cmake [hunk 2]
```diff
@@ -3,10 +3,18 @@
...
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
-set(PICO_SDK_VERSION_MINOR 1)
+set(PICO_SDK_VERSION_MINOR 2)
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 set(PICO_SDK_VERSION_REVISION 0)
+# PICO_BUILD_DEFINE: PICO_SDK_VERSION_PRE_RELEASE_ID, optional SDK pre-release version identifier, type=string, pico_base
+# PICO_CONFIG: PICO_SDK_VERSION_PRE_RELEASE_ID, SDK version revision, type=string, pico_base
+set(PICO_SDK_VERSION_PRE_RELEASE_ID develop)
```
Comment (lurch) [inline pico_sdk_version.cmake:12]: Does this also get displayed by the version-number in the Doxygen output?

Comment (kilograham) [reply pico_sdk_version.cmake:12]: yes

File: pico_sdk_version.cmake [hunk 3]
```diff
@@ -3,10 +3,18 @@
 set(PICO_SDK_VERSION_MAJOR 1)
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_MINOR, SDK minor version number, type=int, pico_base
-set(PICO_SDK_VERSION_MINOR 1)
+set(PICO_SDK_VERSION_MINOR 2)
 # PICO_BUILD_DEFINE: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 # PICO_CONFIG: PICO_SDK_VERSION_REVISION, SDK version revision, type=int, pico_base
 set(PICO_SDK_VERSION_REVISION 0)
+# PICO_BUILD_DEFINE: PICO_SDK_VERSION_PRE_RELEASE_ID, optional SDK pre-release version identifier, type=string, pico_base
+# PICO_CONFIG: PICO_SDK_VERSION_PRE_RELEASE_ID, SDK version revision, type=string, pico_base
```
Comment (lurch) [inline pico_sdk_version.cmake:11]: Oh, I guess the PICO_CONFIG description also needs editing to match the PICO_BUILD_DEFINE description? "optional SDK pre-release version identifier"

And I guess I need to update `extract_configs.py` to also look for `PICO_CONFIG` entries in `*.cmake` files too :grin: (currently it only looks in `*.c` and `*.h` files)

...and for all the "config items" in this file, it looks like `, pico_base` needs correcting to `, group=pico_base`.

Comment (kilograham) [reply pico_sdk_version.cmake:11]: ah yes; this seemed like a better place for them to be in this case

**Comment by lurch** [comment]
You still need to change all the `, pico_base` to `, group=pico_base` but otherwise this LGTM :+1:

**Review by lurch — APPROVED** [review]

**Review by Wren6991 — APPROVED** [review]

**Pull request merged**